src/delivery.cpp

```diff
@@ -0,0 +1,180 @@
+#include <array>
+#include <iostream>
+#include <string>
+#include <vector>
+
+#include "ros/ros.h"
+#include "std_msgs/Bool.h"
+#include "std_msgs/Int8.h"
+#include "std_msgs/String.h"
+
+#include <webots_ros/set_float.h>
+#include <geometry_msgs/Vector3.h>
+
+// For sleep function
+#ifdef _WIN32
+#include <Windows.h>
+#else
+#include <unistd.h>
+#endif
+
+// Hardcoded GPS values
+double placeA[3]{-41.42,0.0,40.82};
+double placeB[3]{-2.3,0,49.45};
+double placeC[3]{58.59,0,40.82};
+double placeD[3]{93.61,0,40.82};
+double placeNil[3]{100000.0,100000.0,100000.0}; // default
+
+bool reachedSetpointBool{0};
+
+std::string senderLocation{};
```

Comment:
One of the worst ways is to create a global variable like this and use it inside all the functions based on your convenience. Naturally, you can always avoid this by creating functions with return types and creating a variable in main and sending the pointer to the functions!